utils/src/activityLog/Activity.ts

```diff
@@ -80,6 +85,7 @@ export abstract class ActivityBase<R> implements hTypes.Activity {
             this._onErrorEmitter.fire({ ...this.getState(), error: e });
             throw e;
         } finally {
+            this.dispose();
```

Comment:
Does `run()` always get run? If so, this object is always disposing of itself, and never needs to be disposed by the caller. In that case I wouldn't make it implement `vscode.Disposable`.

Reply:
Oh shoot good point, run doesn't always get run, I'll remove implementing disposable